### ISSUE TITLE
Customize loss function in a way that ignore the samples with large errors

### ISSUE THREAD

**Issue by vivichina** [NVIDIA/DIGITS#1446]
Hi all，

I have some training data, but some of the labels of them are wrong, so the trained classifier can not achieve good performance, and correct the wrong labels needs to consume a lot of time.

To eliminate the effect of error labels, I intend to redefine the loss function like that: **When calculating the loss of a batch of data, the largest cross-entropy is removed from the others at this batch. And then calculate the mean of the remaining elements as the loss value of this batch.**

I know that digits support python-layer, and if this idea can be achieved, how should I define the loss function via python-layer.
thank you all!

**Comment by lukeyeager** [comment]
We've got this example explaining how to write Python layers:
https://github.com/NVIDIA/DIGITS/blob/digits-5.0/examples/python-layer/README.md

As another example, here's a Python layer implementation of the standard Accuracy layer:
https://gist.github.com/lukeyeager/5ae183044d96ef1540f2

Does that give you enough to work out your custom layer on your own?

**Comment by alpha358** [comment]
vivichina, did this idea proved to be useful ?
